src/ball/mod.rs

```diff
@@ -21,38 +21,20 @@ impl Ball {
     }
 
     fn handle_collisions(&mut self, players: &[&Player; 2]) {
-        players
-            .iter()
+        let ball_vec = na::Vector2::new(self.position.0, self.position.1);
+
+        players.iter()
             .map(|p| p.paddle_vertex())
-            .for_each(|player_paddle| {
-                let left_upper_corner = player_paddle[0];
-                let left_bottom_corner = player_paddle[1];
-                let right_upper_corner = player_paddle[2];
-                let right_bottom_corner = player_paddle[3];
+            .for_each(|p| {
+                if p.chunks(2).any(|c|
+                    dist_to_segment(ball_vec, c[0], c[1]) < 25f32
```

Comment:
Maybe we could remove the magic number (25), and put it into a constant like `BALL_RADIUS`